client/src/lib/attempt-result.js

```diff
@@ -124,6 +139,107 @@ function mean(values) {
   return Math.round(sum / values.length);
 }
 
+/**
+ * return the number of competitors that should be marked as advancing.
+ * When no competitors advance, return 3 for the podium placements.
+ * Might not be accurate for percentage based advancement but that is acceptable
+ */
+function getAdvancingCount(results, advancementCondition) {
+  if (!advancementCondition) return 3;
+  if (advancementCondition.type === "ranking") {
+    return advancementCondition.level;
+  }
+  return results.filter((result) => result.advancing || result.advancingQuestionable).length;
+}
+
+/**
+ * return projectedAverage based on current results.
+ * projections are defined as follows:
+ *   - MO3 events: mean of current solves
+ *   - AVG5 events:
+ *     - 1-2 solves: mean of current solves
+ *     - 3-4 solves: median of current solves
+ */
+export function computeProjectedAverage(result, format) {
+  const attemptResults = result.attempts.map((attempt) => attempt.result);
+  if (result.average) {
+    return result.average;
+  }
+  if (attemptResults.length > 0) {
+    if (format.numberOfAttempts === 3 || attemptResults.length < 3) {
+      return meanOfX(attemptResults);
+    }
+    if (attemptResults.length === 3) {
+      const [, x,] = attemptResults.slice().sort(compareAttemptResults);
+      return x;
+    }
+    const [, x, y,] = attemptResults.slice().sort(compareAttemptResults);
+    return meanOfX([x, y]);
+  }
+  return SKIPPED_VALUE;
+}
+
+/**
+ * return the results object with additional properties:
+ * projectedAverage: projected final average based on current results
+ * forFirst: time needed to overtake first place
+ * forThird: time needed to overtake third place
+ */
+export function getExpandedResults(results, format, forecastView, advancementCondition) {
```

Comment:
Let's move this to `result.js`.

Basically: `attempt-result.js` for various calculations on attempt result (numeric values), and `result.js` for dealing with result objects.

I would also rename this to `resultsForView`.